lib/mirage.ml

```diff
@@ -1237,42 +1202,37 @@ let mprof_trace ~size () =
     method keys = [ Key.abstract key ]
     method packages =
       Key.match_ Key.(value target) @@ function
-      | `Xen | `Qubes -> [package ~sublibs:["xen"] "mirage-profile"]
-      | `Virtio | `Ukvm -> failwith  "tracing is not currently implemented for solo5 targets"
-      | `Unix | `MacOSX -> [package ~sublibs:["unix"] "mirage-profile"]
-
-    method configure _ =
+      | `Xen | `Qubes -> [ package ~sublibs:["xen"] "mirage-profile" ]
+      | `Virtio | `Ukvm -> []
+      | `Unix | `MacOSX -> [ package ~sublibs:["unix"] "mirage-profile" ]
+    method build _ =
       match query_ocamlfind ["lwt.tracing"] with
       | Ok _ -> Ok ()
       | Error _ ->
-        Log.err (fun m ->
-            m "lwt.tracing module not found. Hint:\
-               opam pin add lwt 'https://github.com/mirage/lwt.git#tracing'") ;
-        Error (`Msg "no lwt.tracing")
-
+        R.error_msg "lwt.tracing module not found. Hint:\
+                     opam pin add lwt https://github.com/mirage/lwt.git#tracing"
     method connect i _ _ = match get_target i with
       | `Virtio | `Ukvm -> failwith  "tracing is not currently implemented for solo5 targets"
       | `Unix | `MacOSX ->
         Fmt.strf
           "Lwt.return ())@.\
            let () = (@ \
-             @[<v 2>  let buffer = MProf_unix.mmap_buffer ~size:%a %S in@ \
-             let trace_config = MProf.Trace.Control.make buffer MProf_unix.timestamper in@ \
-             MProf.Trace.Control.start trace_config@]"
+           @[<v 2> let buffer = MProf_unix.mmap_buffer ~size:%a %S in@ \
+           let trace_config = MProf.Trace.Control.make buffer MProf_unix.timestamper in@ \
+           MProf.Trace.Control.start trace_config@]"
           Key.serialize_call (Key.abstract key)
           unix_trace_file;
       | `Xen | `Qubes ->
         Fmt.strf
           "Lwt.return ())@.\
            let () = (@ \
-             @[<v 2>  let trace_pages = MProf_xen.make_shared_buffer ~size:%a in@ \
-             let buffer = trace_pages |> Io_page.to_cstruct |> Cstruct.to_bigarray in@ \
-             let trace_config = MProf.Trace.Control.make buffer MProf_xen.timestamper in@ \
-             MProf.Trace.Control.start trace_config;@ \
-             MProf_xen.share_with (module Gnt.Gntshr) (module OS.Xs) ~domid:0 trace_pages@ \
-             |> OS.Main.run@]"
+           @[<v 2> let trace_pages = MProf_xen.make_shared_buffer ~size:%a in@ \
+           let buffer = trace_pages |> Io_page.to_cstruct |> Cstruct.to_bigarray in@ \
+           let trace_config = MProf.Trace.Control.make buffer MProf_xen.timestamper in@ \
+           MProf.Trace.Control.start trace_config;@ \
+           MProf_xen.share_with (module Gnt.Gntshr) (module OS.Xs) ~domid:0 trace_pages@ \
+           |> OS.Main.run@]"
           Key.serialize_call (Key.abstract key)
```

Comment:
Yes, under Xen we need to grant dom0 permission to read the trace buffer. On Unix, we mmap a file. I think Travis tests that tracing works (or at least compiles).